ext/msgpack/factory_class.c

```diff
@@ -209,6 +211,10 @@ static VALUE Factory_register_type(int argc, VALUE* argv, VALUE self)
         }
     }
 
+    if (ext_module == rb_cInteger && RB_TEST(rb_hash_aref(options, ID2SYM(rb_intern("bigint"))))) {
```

Comment:
No idea what's the best key to pass, so I went with `bigint: true`, but I'm open to suggestions.